services/motion/builtin/move_request.go

```diff
@@ -200,70 +332,54 @@ func (mr *moveRequest) obstaclesIntersectPlan(
 		plan = append(plan, input)
 	}
 
-	for visSrvc, cameraNames := range mr.obstacleDetectors {
-		for _, camName := range cameraNames {
-			mr.logger.Debugf(
-				"proceeding to get detections from vision service: %s with camera: %s",
-				visSrvc.Name().ShortName(),
-				camName.ShortName(),
-			)
-
-			// get detections from vision service
-			detections, err := visSrvc.GetObjectPointClouds(ctx, camName.Name, nil)
-			if err != nil {
-				return state.ExecuteResponse{}, err
-			}
+	// get the already defined geometries of worldstate
+	existingGifs, err := mr.planRequest.WorldState.ObstaclesInWorldFrame(
+		mr.planRequest.FrameSystem,
+		referenceframe.StartPositions(mr.planRequest.FrameSystem),
+	)
+	if err != nil {
+		return state.ExecuteResponse{}, err
+	}
 
-			// Note: detections are initially observed from the camera frame but must be transformed to be in
-			// world frame. We cannot use the inputs of the base to transform the detections since they are relative
+	// existing geometries are in their relative position, i.e. with respect to the base's frame
+	// here we transform them into their absolute positions. i.e. with respect to the world frame
+	// Note: we remove any existing transient obstacle geometries as they will be re-detected
+	existingGeomsAbs := []spatialmath.Geometry{}
+	for _, g := range existingGifs.Geometries() {
+		if strings.Contains(g.Label(), "_transientObstacle_") {
```

Comment:
We should at least throw a warning here; we don't want to silently skip obstacles.